desktop-widgets/locationinformation.cpp

```diff
@@ -313,15 +313,18 @@ void LocationInformationWidget::on_diveSiteDistance_textChanged(const QString &s
 
 void LocationInformationWidget::reverseGeocode()
 {
-	dive_site *ds = diveSite; /* Use local copy in case the object-copy is NULL'd by another thread while we perform the geo lookup */
+	bool buttonWasEnabled = ui.geoCodeButton->isEnabled();
+	ui.geoCodeButton->setEnabled(false);
+	dive_site *ds = diveSite; /* Save local copy; possibility of user closing the widget while reverseGeoLookup is running (see #2930) */
 	location_t location = parseGpsText(ui.diveSiteCoordinates->text());
-	if (!ds || !has_location(&location))
-		return;
 	taxonomy_data taxonomy = { 0, 0 };
-	reverseGeoLookup(location.lat, location.lon, &taxonomy);
-	if (ds != diveSite)
-		return;
-	Command::editDiveSiteTaxonomy(ds, taxonomy);
+	if (ds && has_location(&location)) {
+		reverseGeoLookup(location.lat, location.lon, &taxonomy);
+	}
```

Comment:
Per coding guide, we don't use braces for single line code blocks ... *shrug*

Reply:
That is (IMHO) a rather poor coding guideline.  Agreed, the braces do add an extra line of code, but poterntially prevent coding mistakes. I recently encountered a logic error precisely because of this code style in another project (admittedly the source file also had broken indentation and other issues, but still).

That said, I will, of course, adhere to the project guidelines and change it if I can't convince you to change the guidelines ;)